File: pythatriple-web-api/src/main/java/com/example/pythatriple_web_api/dto/PythatripleRequest.java
package com.example.pythatriple_web_api.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

public record PythatripleRequest(
	@NotNull(message = "parameter is required")
	@Positive(message = "must be a positive integer")
	Integer hypotenuse_squared
) {}

File: pythatriple-web-api/src/main/kotlin/com/example/pythatriple_web_api/dto/PythatripleRequest.kt
package com.example.pythatriple_web_api.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive

@JvmRecord
data class PythatripleRequest(
	@field:NotNull(message = "parameter is required")
	@field:Positive(message = "must be a positive integer")
	val hypotenuse_squared: Int?
)
